__tests__/ExpensiMark-HTML-test.js

```diff
@@ -664,3 +664,161 @@ test('Test for link with no content', () => {
     const resultString = '[  ](<a href="https://www.link.com" target="_blank" rel="noreferrer noopener">www.link.com</a>)';
     expect(parser.replace(testString)).toBe(resultString);
 });
+
+// Valid text that should match for user mentions
+test('Test for user mention with @username@domain.com', () => {
+    const testString = '@username@expensify.com';
+    const resultString = '<mention-user>@username@expensify.com</mention-user>';
+    expect(parser.replace(testString)).toBe(resultString);
+});
+
+test('Test for user mention with @phoneNumber@domain.sms', () => {
+    const testString = '@+19728974297@expensify.sms';
+    const resultString = '<mention-user>@+19728974297@expensify.sms</mention-user>';
+    expect(parser.replace(testString)).toBe(resultString);
+});
+
+test('Test for user mention with bold style', () => {
+    const testString = '*@username@expensify.com*';
+    const resultString = '<strong><mention-user>@username@expensify.com</mention-user></strong>';
+    expect(parser.replace(testString)).toBe(resultString);
+});
+
+test('Test for user mention with italic style', () => {
+    const testString = '_@username@expensify.com_';
+    const resultString = '<em><mention-user>@username@expensify.com</mention-user></em>';
+    expect(parser.replace(testString)).toBe(resultString);
+});
+
+test('Test for user mention with heading1 style', () => {
+    const testString = '# @username@expensify.com';
+    const resultString = '<h1><mention-user>@username@expensify.com</mention-user></h1>';
+    expect(parser.replace(testString)).toBe(resultString);
+});
+
+test('Test for user mention with strikethrough style', () => {
+    const testString = '~@username@expensify.com~';
+    const resultString = '<del><mention-user>@username@expensify.com</mention-user></del>';
+    expect(parser.replace(testString)).toBe(resultString);
+});
+
+test('Test for user mention with @here', () => {
+    const testString = '@here say hello to @newuser@expensify.com';
+    const resultString = '<mention-here>@here</mention-here> say hello to <mention-user>@newuser@expensify.com</mention-user>';
+    expect(parser.replace(testString)).toBe(resultString);
+});
+
+/**
```

Comment:
IMO we should remove all this. We don't need to leave a bunch of commented-out code sitting in the codebase waiting for it to be implemented.